### ISSUE TITLE
nice to have: ssh-to-gpg available in shell.nix

### ISSUE THREAD

**Issue by colemickens** [Mic92/sops-nix#29]
It would be nice to have `ssh-to-gpg` exposed easily from `shell.nix`.

But then again, maybe the goal is to delete it once `age` is merged into `sops` and you or someone else adds sshpubkey support to `sops`?

**Comment by Mic92** [comment]
Do you mean here? https://github.com/Mic92/sops-nix/blob/master/shell.nix

**Comment by colemickens** [comment]
Yeah, that would make it convenient for users to do `nix shell github:mic92/sops-nix#ssh-to-gpg -p ssh-to-gpg -i ...` or whatever the exact flake syntax is without having cloned the repo and know to do `go run ...`, etc.

**Comment by Mic92** [comment]
In that case I need to add it to `flake.nix`.

**Comment by Mic92** [comment]
It is also exposed here `https://github.com/Mic92/sops-nix/blob/master/default.nix#L8`. I think `nix-shell` also allows to download it from there.

**Comment by colemickens** [comment]
Sorry, I'm not explaining myself well.

Scenario: I'm writing a tutorial. Or I'm following the tutorial on this repo's README. I don't necessarily want to develop `sops-nix` and clone the repo. How do I run `ssh-to-gpg` without cloning this repo, or writing my own shell.nix?

>In that case I need to add it to packages in flake.nix.

Maybe that's what I should've asked for. I have my shell.nix load my devShell which I think then loads all of my packages into it for this sort of scenario.

**Comment by Mic92** [comment]
I think I understood your intention, how about this line?

```console
$ nix-shell -E 'with import <nixpkgs> {}; pkgs.mkShell { buildInputs = [ (pkgs.callPackage (fetchTarball https://github.com/Mic92/sops-nix/archive/master.tar.gz) {}).ssh-to-pgp]; }'
```

**Comment by Mic92** [comment]
With flakes you can already do:

```
nix run  'github:Mic92/sops-nix#ssh-to-pgp'
```

**Comment by colemickens** [comment]
Oh. I'm still bad at getting flake outputs right. Sorry for the distraction. That's what I wanted. `nix run 'github:Mic92/sops-nix#ssh-to-pgp'`

**Comment by Mic92** [comment]
It still complains about lock files not beeing up-to-date. I am fixing this here: https://github.com/Mic92/sops-nix/pull/31
